### PR TITLE
Clean_up related to symex_assign_symbol

### PR DESCRIPTION
**Pull request by romainbrenguier**
There is no functional changes, this is just aimed at making the code a bit simpler

- [ ] Each commit message has a non-empty body, explaining why the change was made.
- [na] Methods or procedures I have added are documented, following the guidelines provided in CODING_STANDARD.md.
- [na] The feature or user visible behaviour I have added or modified has been documented in the User Guide in doc/cprover-manual/
- [ ] Regression or unit tests are included, or existing tests cover the modified code (in this case I have detailed which ones those are in the commit message).
- [na] My commit message includes data points confirming performance improvements (if claimed).
- [ ] My PR is restricted to a single feature or bugfix.
- [na] White-space or formatting changes outside the feature-related changed lines are in commits of their own.

### CODE REVIEW

**Review by allredj — COMMENTED** [review]
:heavy_check_mark:
Passed Diffblue compatibility checks (cbmc commit: bc044d93300232080331ce144a807ef78a7981bd).
Build URL: https://travis-ci.com/diffblue/test-gen/builds/115563600

**Review by owen-mc-diffblue — APPROVED** [review]

**Review by peterschrammel — APPROVED** [review]

File: src/goto-symex/symex_assign.cpp [hunk 1]
```diff
@@ -434,16 +434,10 @@ void goto_symext::symex_assign_symbol(
...
+  exprt l2_rhs = state.rename(
     // put assignment guard into the rhs
-    if(!guard.empty())
-    {
-      guarded_rhs =
-        if_exprt{conjunction(guard), std::move(guarded_rhs), lhs, rhs.type()};
-    }
-    return state.rename(std::move(guarded_rhs), ns).get();
-  }();
+    guard.empty() ? rhs : (exprt)if_exprt{conjunction(guard), rhs, lhs}, ns)
```
Comment (tautschnig) [inline src/goto-symex/symex_assign.cpp]: I'd prefer `static_cast<exprt>` as the C-style cast is just very easy to overlook.

File: src/goto-symex/ssa_step.cpp [hunk 2]
```diff
@@ -231,7 +230,7 @@ SSA_assignment_stept::SSA_assignment_stept(
   original_full_lhs = std::move(_original_full_lhs);
   ssa_rhs = std::move(_ssa_rhs);
   assignment_type = _assignment_type;
-  cond_expr = std::move(_cond_expr);
+  cond_expr = equal_exprt(ssa_lhs, ssa_rhs);
```
Comment (tautschnig) [inline src/goto-symex/ssa_step.cpp]: Nit pick: use curly braces

File: src/goto-symex/symex_assign.cpp [hunk 3]
```diff
@@ -475,15 +475,11 @@ void goto_symext::symex_assign_symbol(
               << messaget::eom;
     });
 
-  const exprt assignment_guard =
-    make_and(state.guard.as_expr(), conjunction(guard));
-
-  const exprt original_lhs = get_original_name(l2_full_lhs);
   target.assignment(
```
Comment (tautschnig) [inline src/goto-symex/symex_assign.cpp:483]: Random place: typo in the commit message: "this values" -> "these values", and also it should be "Inline variables" (plural)

File: src/goto-symex/goto_state.cpp [hunk 4]
```diff
@@ -31,19 +31,17 @@ std::size_t goto_statet::increase_generation(
...
+  const std::size_t n = fresh_l2_name_provider(l1_identifier);
 
-  const auto r_opt = level2.current_names.find(l1_identifier);
-
-  if(!r_opt)
+  if(const auto r_opt = level2.current_names.find(l1_identifier))
   {
-    level2.current_names.insert(l1_identifier, std::make_pair(lhs, n));
+    std::pair<ssa_exprt, unsigned> copy = r_opt->get();
+    copy.second = narrow<unsigned>(n);
```
Comment (tautschnig) [inline src/goto-symex/goto_state.cpp]: Ack, but we should really stop using `unsigned`, the renaming index ought to be `std::size_t`.

File: src/goto-symex/goto_symex_state.h [hunk 5]
```diff
@@ -217,7 +217,7 @@ class goto_symex_statet final : public goto_statet
   unsigned remaining_vccs = 0;
 
   /// Allocates a fresh L2 name for the given L1 identifier, and makes it the
-  //  latest generation on this path.
+  /// latest generation on this path.
```
Comment (tautschnig) [inline src/goto-symex/goto_symex_state.h:220]: Note on the commit message: that's a "slash", not a "backslash"

**Review by tautschnig — APPROVED** [review]

File: src/goto-symex/goto_state.cpp (hunk 4 above)
Comment (romainbrenguier) [reply src/goto-symex/goto_state.cpp]: @tautschnig I added that change in a new commit

Comment (tautschnig) [reply src/goto-symex/goto_state.cpp]: Thank you!

**Review by allredj — COMMENTED** [review]
:heavy_check_mark:
Passed Diffblue compatibility checks (cbmc commit: 71caf32f40eb7701d1072807d19ca19991d86955).
Build URL: https://travis-ci.com/diffblue/test-gen/builds/115789758

**Pull request merged**